### ISSUE TITLE
Steward incompatible to ArgoCD 1.8

### ISSUE THREAD

**Issue by megian** [projectsyn/steward#61]
Because the ArgoCD  projectsyn/component-argocd#14 1.8 upgrade  requires switching from a deployment to a statefulset, Steward now tries to recreate the deployment as it believes ArgoCD is not fully deployed. 

```
I0204 16:25:04.214181       1 argocd.go:52] Found 3 of expected 4 deployments, bootstrapping now
I0204 16:25:04.231714       1 configmap.go:97] Update existing ConfigMap argocd-ssh-known-hosts-cm
I0204 16:25:04.252839       1 configmap.go:97] Update existing ConfigMap argocd-tls-certs-cm
I0204 16:25:04.282520       1 configmap.go:97] Update existing ConfigMap argocd-rbac-cm
I0204 16:25:04.296355       1 configmap.go:97] Update existing ConfigMap argocd-cm
I0204 16:25:04.431858       1 argo-crds.go:62] applications.argoproj.io CRD already exists, skip
I0204 16:25:04.486824       1 argo-crds.go:62] appprojects.argoproj.io CRD already exists, skip
W0204 16:25:04.527691       1 redis.go:86] Argo CD redis service already exists
W0204 16:25:04.556087       1 redis.go:95] Argo CD redis deployment already exists
W0204 16:25:04.856737       1 repo-server.go:137] Argo CD repo-server service already exists
W0204 16:25:04.970265       1 repo-server.go:146] Argo CD repo-server deployment already exists
W0204 16:25:05.090080       1 server.go:130] Argo CD server already exists
W0204 16:25:05.271566       1 argo-app.go:60] Argo Project already exists, skip
W0204 16:25:05.340621       1 argo-app.go:106] Argo App already exists, skip
I0204 16:25:05.358146       1 app-controller.go:103] Created Argo CD application-controller deployment
```

**Comment by megian** [comment]
PR #62

**Comment by megian** [comment]
Require changing the push/release workflow #70 before releasing the new version.

**Comment by megian** [comment]
The upgrade requires to update the argocd component and steward simultaneously:
```
parameters:
  component_versions:
    argocd:
      version: upgrade-1.8

  steward:
    images:
      steward:
        tag: 'latest'
```

**Comment by megian** [comment]
Released the version v0.3.0.

The upgrade requires to update the argocd component and steward simultaneously:
```
parameters:
  component_versions:
    argocd:
      version: upgrade-1.8

  steward:
    images:
      steward:
        tag: 'v0.3.0@sha256:457532633fb680fa58524814c45401705299ba14f96354e61d90d26aaa2b8a9b'
```

**Comment by megian** [comment]
Released the version v0.3.1 with some important bug fixes.

The upgrade requires to update the argocd component and steward simultaneously:
```
parameters:
  component_versions:
    argocd:
      version: upgrade-1.8

  steward:
    images:
      steward:
        tag: 'v0.3.1@sha256:066129ff4e932cd1f06097f637c6b2a993454f5bf0862ba530d8439ceb629129'
```
